### PR TITLE
[Form recognizer]Add validation checks before setting valueData (#22057)

### PR DESCRIPTION
**Pull request by samvaity**
Applying the hotfix branch changes to the main branch.
Ref - https://github.com/Azure/azure-sdk-for-java/pull/22057

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md [hunk 1]
```diff
@@ -478,7 +486,7 @@ FormRecognizerAsyncClient formRecognizerAsyncClient = new FormRecognizerClientBu
 
 ### Additional documentation
 
-For more extensive documentation on Azure Cognitive Services Form Recognizer, see the [Form Recognizer documentation][api_reference_doc].
+For more extensive documentation on Azure Cognitive Services Form Recognizer, see the [Form Recognizer documentation][github_io_docs].
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/README.md:489]: github_io_docs link?

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/Transforms.java [hunk 2]
```diff
@@ -305,13 +308,12 @@ private static TextAppearance getTextAppearance(TextLine textLine) {
                 if (fieldValue != null) {
                     List<FormElement> formElementList = setReferenceElements(fieldValue.getElements(), readResults);
                     FieldData valueData;
-                    // Bounding box, page and text are not returned by the service in two scenarios:
+                    // Bounding box and page an are not returned by the service in two scenarios:
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/Transforms.java:311]: bounding box and page?

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/Transforms.java [hunk 3]
```diff
@@ -305,13 +308,12 @@ private static TextAppearance getTextAppearance(TextLine textLine) {
...
                     //   - When this field is global and not associated with a specific page (e.g. ReceiptType).
                     //   - When this field is a collection, such as a list or dictionary.
                     //
                     // In these scenarios we do not set a ValueData.
-                    if (fieldValue.getText() == null && fieldValue.getPage() == null
-                        && CoreUtils.isNullOrEmpty(fieldValue.getBoundingBox())) {
+                    if (fieldValue.getPage() == null && CoreUtils.isNullOrEmpty(fieldValue.getBoundingBox())) {
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/Transforms.java:316]: if fieldValue.getText() == null, do we create an null value FieldData and possible having BoundingBox?

Comment (samvaity) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/Transforms.java:316]: Yes, we create a FieldData with null text but values for page and BoundingBox.

**Review by mssfang — APPROVED** [review]

**Pull request merged**